src/custom/state/swap/hooks.ts

```diff
@@ -0,0 +1,224 @@
+import { getTip } from '@src/custom/utils/price'
+import { useV1Trade } from '@src/data/V1'
+import { useActiveWeb3React } from '@src/hooks'
+import { useCurrency } from '@src/hooks/Tokens'
+import { useTradeExactIn, useTradeExactOut } from '@src/hooks/Trades'
+import useENS from '@src/hooks/useENS'
+import useToggledVersion, { Version } from '@src/hooks/useToggledVersion'
+import { Field } from '@src/state/swap/actions'
+import { tryParseAmount, useSwapState } from '@src/state/swap/hooks'
+import { useUserSlippageTolerance } from '@src/state/user/hooks'
+import { useCurrencyBalances } from '@src/state/wallet/hooks'
+import { isAddress } from '@src/utils'
+import { computeSlippageAdjustedAmounts } from '@src/utils/prices'
+import { Currency, CurrencyAmount, JSBI, Token, TokenAmount, Trade } from '@uniswap/sdk'
+
+export * from '@src/state/swap/hooks'
+
+const BAD_RECIPIENT_ADDRESSES: string[] = [
+  '0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f', // v2 factory
+  '0xf164fC0Ec4E93095b804a4795bBe1e041497b92a', // v2 router 01
+  '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D' // v2 router 02
+]
+
+/**
+ * Returns true if any of the pairs or tokens in a trade have the given checksummed address
+ * @param trade to check for the given address
+ * @param checksummedAddress address to check in the pairs and tokens
+ */
+function involvesAddress(trade: Trade, checksummedAddress: string): boolean {
+  return (
+    trade.route.path.some(token => token.address === checksummedAddress) ||
+    trade.route.pairs.some(pair => pair.liquidityToken.address === checksummedAddress)
```

Comment:
Will these `token.address` and `liquidityToken.address` always be checksummed? or maybe we need `toLowerCase()`

Reply:
hmm good question but I wouldnt focus on changes I haven't made. You can't see them as the entire file is copied but I didn't touch this.

Though maybe your point is more that maybe they have some service we're not aware of happening outside of their code here that is returning consistent checksummed addresses